### DISCUSSION TITLE
Questions on Tone Mapping

### DISCUSSION THREAD

**Discussion by tabetVercetti** [pmndrs/postprocessing#558] [Q&A]
Hello all :) im enjoying this wonderful postprocessing library and i would like to better understand certain aspects of it. At the moment I am learning how to use tonemappingeffect. But i have some questions that i hope the community can help me with:

1- I understand that ToneMappingEffect should be the last pass I add to my composer. What are the recommendations when I have a selectivebloom pass which happens before in my pipeline. At the moment i had a red object blooming, but the tonemapping turns it into a light red/orange color. I remember this was easily solved in threejs's default tonemapping by simply setting the material parameter "tonemapped" to false. I have done the same here but nothing has changed. Is there a special way to add exceptions to the tonemapping in this library?

2- The default tonemapping in three js was more straight forward. For example, it was enough to select the tonemapping mode to "ACESFilmic" and I would get a good result. But in this library, the tonemappingeffect has extra parameters such as blendFunction/resolution/whitepoint/middleGrey/minLuminance/avgLuminance/adaptationRate. Can someone guide me through how i should set these parameters in order to get the "ACESFilmic" tonemapping that threejs usually does?

**Comment by vanruesc** [comment]
The readme contains information regarding this topic:
https://github.com/pmndrs/postprocessing?tab=readme-ov-file#tone-mapping

The `ToneMappingEffect` uses `ToneMappingMode.ACES_FILMIC` by default. The other settings you mentioned only apply to the other tone mapping modes. More infos can be found in the [docs](https://pmndrs.github.io/postprocessing/public/docs/class/src/effects/ToneMappingEffect.js~ToneMappingEffect.html).
